### ISSUE TITLE
Unsuccessful HTR

### ISSUE THREAD

**Issue by sk-dataocean-online** [mittagessen/kraken#711]
Hi.
I was trying to train a model on a small sample of 700 lines of handwritten text. ketos train linearly increased the accuracy from 0 to less than 20% in 35 epochs which was suspicious. The "best" model in eScriptorium produces almost the same string for all the lines to be recognized. So no recognition was made just random guess. Maybe it is related to the issue in #571 

Could you please reproduce my training and confirm whether it is kraken issue or the dataset issue?

OS: Linux Mint 22.1 (Ubuntu 24.04 LTS)
Python 3.12
Kraken 5.3.0
Kraken was installed in venv by pip install kraken
Dataset was obtained by manual transcription in eScriptorium, downloaded as ALTO with images, then compiled into binary with ketos compile -F output.txt -f alto (where output.txt has a list of XML files).
Training was simple: ketos train -f binary dataset.arrow

[dataset.arrow.zip](https://github.com/user-attachments/files/20942714/dataset.arrow.zip)

**Comment by johnlockejrr** [comment]
You can't train from scratch as you did with such a small amount of data. Try to finetune a model, you'll get better results. What language? What script? You uploaded the binary arrow...

**Comment by sk-dataocean-online** [comment]
Thanks for your reply. What is the minimal volume of data to train from scratch? I realize that the model will be less than moderate from such a small amount of data, but I would like to test the feasibility of such OCR before investing into time-consuming manual transcription. I thought that it should be at least 60-70% accuracy from my sample but not just random guess which is dummy model.

It is Russian handwriting church records of 19th century.

Example of text: 

![Image](https://github.com/user-attachments/assets/7b3a45da-3134-4c80-b7d6-0444be96558b)

**Comment by sk-dataocean-online** [comment]
Thank you so much for your reply! Indeed, fine-tuning of CATMuS yielded acceptable results - 86% accuracy after 12th epoch then stuck around. So it is just about dataset size. According to your experience, what dataset size for handwriting will be acceptable to train from scratch?

**Comment by mittagessen** [comment]
On 25/06/27 07:08AM, sk-dataocean-online wrote:
> sk-dataocean-online left a comment (mittagessen/kraken#711)
> 
> Thank you so much for your reply! Indeed, fine-tuning of CATMuS
> yielded acceptable results - 86% accuracy after 12th epoch then stuck
> around. So it is just about dataset size. According to your
> experience, what dataset size for handwriting will be acceptable to
> train from scratch?

Even with very large datasets it can often be advantegeous to fine-tune
from an existing model, if only to reduce training times. With your
fairly substantial base dataset size and your material being far out of
domain to the original CATMuS dataset you don't even risk the model
retaining some bad habits from the data the base model has been trained
on. 

The required dataset size depends on what you want to do, the complexity
of the writing, and complexity of the chosen transcription guidelines.
Let's say writing in your script or the document style you want to
recognize is very regular, has a small glyph inventory, not much
orthographic variation, and you don't require exceptionally high
character accuracy (<1%), you'll be able to get away with a couple of
thousand well-sampled lines. If you want a model that recognizes a
wide-range of handwriting styles and/or print in a script with lots of
different glyphs and you use a very diplomatic transcription you'll
require substantially more data.

Data quality is also important. People often create datasets by
transcribing a whole document manually which will limit the
generalization of the model in many cases. It is better to create a
dataset of a 100 pages from 1 page taken from a 100 different documents
each than by transcribing 100 pages from a single one.

Lastly, creating good general models for historical material requires a
bit of paleographical expertise which might not even exist for
the kind of material you're working on.
